### PR TITLE
CT-111: remove version param from progress bubble nav url

### PR DESCRIPTION
**Pull request by Erin007**
When viewing a previous version of a project-back level and then navigating to a level in that is neither backed by a project nor backed by the same project the version id in the url causes a confusing error state because the version id is project-specific. 

Screenshot 2023-10-06 at 10 18 23 AM

To alleviate this confusion, we're now removing the version param if it exists when generating the level url for a bubble.

There is a slight downside to this change in cases where users are navigating between levels that backed by the _same_ project; they could go from an old version of a project in one level to the most recent version of a project in a different level.  However, this change felt least complex to solve the majority of likely use cases and there is already a handy banner visually indicating to a user when they are viewing a project that is not the latest version. 

## Links

 [CT-111](https://codedotorg.atlassian.net/browse/CT-111)

## Testing story

Manually confirmed that expected behavior locally. 
BEFORE 

https://github.com/code-dot-org/code-dot-org/assets/12300669/bba1f1f7-2fa1-48c5-b1e0-1068762fed19

AFTER 

https://github.com/code-dot-org/code-dot-org/assets/12300669/1e314ead-48fe-41a7-9587-e7bc5f0845f3

And added a unit test for this particular usage of `getBubbleUrl`

## Deployment strategy
No special consideration needed. 

## Privacy

No. 
## Security

n/a

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by molly-moen — COMMENTED** [review]
Change looks good, looks like the new test is failing on drone though 😢

**Comment by Erin007** [comment]
> Change looks good, looks like the new test is failing on drone though 😢

oh that's fun. I love when a test passes locally and fails drone 🙃 I think I've fixed it, though 🤞

**Review by molly-moen — APPROVED** [review]
LGTM!

**Review by bethanyaconnor — APPROVED** [review]

File: apps/test/unit/templates/progress/BubbleFactoryTest.jsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
 import {shallow, mount} from 'enzyme';
+import {updateQueryParam} from '../../../../src/code-studio/utils';
+import {currentLocation} from '@cdo/apps/utils';
```
Comment (mikeharv) [inline apps/test/unit/templates/progress/BubbleFactoryTest.jsx:4]: Q: I've often been curious about these two different import path formats. Would `@cdo/apps/utils/code-studio/utils` have worked for `updateQueryParam`? I've tended to use the shortcut path when I see lots of repeating `../` in the relative path, but I'm not sure if there are other schools of thought.

**Review by mikeharv — APPROVED** [review]
Looks good! Thanks for including the helpful before and after videos!

**Comment by Erin007** [comment]
Product gave this a 👍 so it's ready to go!

File: apps/test/unit/templates/progress/BubbleFactoryTest.jsx (hunk 1 above)
Comment (Erin007) [reply apps/test/unit/templates/progress/BubbleFactoryTest.jsx:4]: Good question! I know this has come up in the past (e.g. https://codedotorg.slack.com/archives/C0T0PNR0D/p1563236491128200) and as far as I know we haven't pushed for or landed on standardization. I was trained under the old regime though so if someone's interested in championing consistency here I can get on board!

**Pull request merged**